src/org/intellij/erlang/ErlangReferenceContributor.java

```diff
@@ -29,30 +31,39 @@
 
   @Override
   public void registerReferenceProviders(@NotNull PsiReferenceRegistrar registrar) {
-    register(registrar, Kind.RECORD,   "erlang", "is_record",  2, 1);
-    register(registrar, Kind.RECORD,   "erlang", "is_record",  3, 1);
-    register(registrar, Kind.MODULE,   "erlang", "spawn",      3, 0);
-    register(registrar, Kind.FUNCTION, "erlang", "spawn",      3, 1);
-    register(registrar, Kind.MODULE,   "erlang", "spawn_link", 3, 0);
-    register(registrar, Kind.FUNCTION, "erlang", "spawn_link", 3, 1);
+    registerRecords(registrar, "erlang", "is_record", 2, 1);
+    registerRecords(registrar, "erlang", "is_record", 3, 1);
+
+    ErlangFunctionCallNamedParameter<PsiElement> moduleParameter =
+      new ErlangFunctionCallNamedParameter<PsiElement>("Module", "Mod", "_Module");
+    registrar.registerReferenceProvider(psiElement(ErlangQAtom.class).with(moduleParameter), getProvider(Kind.MODULE));
+
+    ErlangFunctionCallNamedParameter<PsiElement> funParameter =
+      new ErlangFunctionCallNamedParameter<PsiElement>("Function", "Fun", "_Function");
+    registrar.registerReferenceProvider(psiElement(ErlangQAtom.class).with(funParameter), getProvider(Kind.FUNCTION));
```

Comment:
It won't work really well, as you will only create a reference to function name, which will only resolve to local function declarations - you can do something better here, I believe.